src/town.h

```diff
@@ -6,60 +6,27 @@
 
 #include "position.h"
 
-class Town
+struct Town
 {
-public:
-	explicit Town(uint32_t id) : id(id) {}
-
-	const Position& getTemplePosition() const { return templePosition; }
-	const std::string& getName() const { return name; }
-
-	void setTemplePos(Position pos) { templePosition = pos; }
-	void setName(std::string_view name) { this->name = name; }
-	uint32_t getID() const { return id; }
-
-private:
-	uint32_t id;
+	const uint32_t id;
 	std::string name;
 	Position templePosition;
 };
 
-using TownMap = std::map<uint32_t, Town*>;
+using TownMap = std::map<uint32_t, std::unique_ptr<Town>>;
 
 class Towns
 {
 public:
-	Towns() = default;
-	~Towns()
-	{
-		for (const auto& it : townMap) {
-			delete it.second;
-		}
-	}
-
-	// non-copyable
-	Towns(const Towns&) = delete;
-	Towns& operator=(const Towns&) = delete;
-
-	bool addTown(uint32_t townId, Town* town) { return townMap.emplace(townId, town).second; }
-
-	Town* getTown(const std::string& townName) const
-	{
-		for (const auto& it : townMap) {
-			if (caseInsensitiveEqual(townName, it.second->getName())) {
-				return it.second;
-			}
-		}
-		return nullptr;
-	}
+	void setTown(uint32_t townId, Town* town) { townMap[townId].reset(town); }
 
-	Town* getTown(uint32_t townId) const
+	const Town* getTown(uint32_t townId) const
 	{
 		auto it = townMap.find(townId);
 		if (it == townMap.end()) {
 			return nullptr;
 		}
-		return it->second;
+		return it->second.get();
 	}
 
 	const TownMap& getTowns() const { return townMap; }
```

Comment:
We could use auto and keep the type inferred directly from the townMap variable ;)

Reply:
I'd rather do it separately. Probably would change to `decltype(auto) getTowns() const` to avoid copying